homeassistant/components/tahoma/coordinator.py

```diff
@@ -0,0 +1,150 @@
+"""Helpers to help coordinate updates."""
+from datetime import timedelta
+import logging
+from typing import Dict, List, Optional, Union
+
+from aiohttp import ServerDisconnectedError
+from pyhoma.client import TahomaClient
+from pyhoma.enums import EventName, ExecutionState
+from pyhoma.exceptions import (
+    BadCredentialsException,
+    MaintenanceException,
+    NotAuthenticatedException,
+    TooManyRequestsException,
+)
+from pyhoma.models import DataType, Device, State
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+TYPES = {
+    DataType.NONE: None,
+    DataType.INTEGER: int,
+    DataType.DATE: int,
+    DataType.STRING: str,
+    DataType.FLOAT: float,
+    DataType.BOOLEAN: bool,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TahomaDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching TaHoma data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        *,
+        name: str,
+        client: TahomaClient,
+        devices: List[Device],
+        update_interval: Optional[timedelta] = None,
+    ):
+        """Initialize global data updater."""
+        super().__init__(
+            hass,
+            logger,
+            name=name,
+            update_interval=update_interval,
+        )
+
+        self.data = {}
+        self.original_update_interval = update_interval
+        self.client = client
+        self.devices: Dict[str, Device] = {d.deviceurl: d for d in devices}
+        self.executions: Dict[str, Dict[str, str]] = {}
+
+        _LOGGER.debug(
```

Comment:
We don't want side effects like logging in init methods.

Reply:
@MartinHjelmare ok, I thought that extra logging, as long as its debug, is possible. What would be the best way to add debug logging (thus no info / warning). It is really helpful to understand someones update interval while investigating authentication issues. 

Do you know any other integrations that do this in the right way? :)

Reply:
We can log the updated interval after creating the coordinator outside it, or after updating the interval.